lua/conform/runner.lua

```diff
@@ -370,13 +370,48 @@ local function run_formatter(bufnr, formatter, config, ctx, input_lines, opts, c
 
   if not config.stdin then
     log.debug("Creating temp file %s", ctx.filename)
+    -- ctx.filename can contain directories relative to ctx.dirname
+    -- Thus, we check which directories relative to ctx.dirname exist and remove
+    -- the first one that does not exist.
+    local res = vim
+      .iter(
+        vim.split(
+          assert(
+            vim.fs.relpath(ctx.dirname, ctx.filename),
+            "filename " .. ctx.filename .. " is not a child of dirname " .. ctx.dirname
+          ),
+          "/",
```

Comment:
Splitting by `/` might not work on windows. Could we do this maybe procedurally or recursively using `vim.fs.dirname` to find the first parent dir that does exist?